File: after_jg/week_1/1966(프린터큐).js
let fs = require('fs');
let input = fs.readFileSync('input.txt').toString().trim().split('\n');

// let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const printer = (list) => {
  let result = []

  while (list.length > 0) {
    let docs = list.shift();
    let [docs_rank, docs_idx] = docs;

    if (list.some((elem) => docs_rank < elem[0])) {
      list.push(docs)
    } else {
      result.push(docs)
    }
  }
  return result;
}

let testCase = parseInt(input[0]);
let cur = 1;
while (cur < testCase * 2) {
  let [N, M] = input[cur].split(' ').map(Number);
  let rank = input[cur + 1].split(' ').map(Number);
  rank = rank.map((elem, idx) => [elem, idx])
  let result = printer(rank);
  console.log(result)
  console.log(result.findIndex((elem) => elem[1] == M) + 1)
  cur += 2
}

File: after_jg/week_1/1966(프린터큐).py

import sys
from collections import deque
input = sys.stdin.readline

T = int(input())
result = []

for _ in range(T):
    N, find_idx = map(int, input().split())
    importances = list(map(int, input().split()))
    count = 0
    printer = deque() 

    for idx, importance in enumerate(importances):
        printer.append([importance, idx])

    while len(printer) > 0 :
        now = printer.popleft()
        is_max = True
        
        for docs in printer:
            if docs[0] > now[0]:
                printer.append(now)
                is_max = False
                break
        
        if is_max:
            count += 1
            if now[1] == find_idx:
                result.append(count)
                break

for res in result:
    print(res)
